File: vm/luavm/coin.lua
-- require module
local ZIP = require("ZIP")

-- 合约创建时会被调用一次，之后就不会被调用
function Init(args)
    print("init...", args)
    return true
end

-- invoke when exec transaction
function Invoke(func, args)
    print("in Invoke", func, args)
    return true
end

-- query
function Query(args)
    return "query ok"
end

File: vm/jsvm/coin.js
/***** JS example *****/

// only exec once when deploy contract
function Init(args) {
    console.log("init", args)
    return true;
}

// to trigger invoke when exec contract
function Invoke(func, args) {
    console.log("invoke", func, args)
    return true;
}

// to trigger query when exec query
function Query(args) {
    console.log("call Query", args);
    return "query ok"
}
